src/keystore.h

```diff
@@ -112,14 +117,15 @@ typedef std::map<
 typedef std::map<
     libzcash::SaplingIncomingViewingKey,
     libzcash::SaplingExtendedFullViewingKey> SaplingFullViewingKeyMap;
-// Only maps from default addresses to ivk, may need to be reworked when adding diversified addresses. 
+// Only maps from default addresses to ivk, may need to be reworked when adding diversified addresses.
 typedef std::map<libzcash::SaplingPaymentAddress, libzcash::SaplingIncomingViewingKey> SaplingIncomingViewingKeyMap;
 
 /** Basic key store, that keeps keys in an address->secret map */
 class CBasicKeyStore : public CKeyStore
 {
 protected:
-    HDSeed hdSeed;
+    std::optional<MnemonicSeed> mnemonicSeed;
```

Comment:
```suggestion
    // All wallets will have a mnemonic seed, but this field must be declared
    // as optional to avoid the need to construct an invalid seed before the
    // wallet's contents have been loaded from the database.
    std::optional<MnemonicSeed> mnemonicSeed;
```